packages/edit-post/src/components/sidebar/post-link/index.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { PanelBody, TextControl } from '@wordpress/components';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { compose, ifCondition, withState } from '@wordpress/compose';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Module Constants
+ */
+const PANEL_NAME = 'post-link';
+
+function PostLink( {
+	isOpened,
+	onTogglePanel,
+	isEditable,
+	postLink,
+	permalinkParts,
+	editPermalink,
+	postType,
+	forceEmptyField,
+	setState,
+} ) {
+	const { prefix, postName, suffix } = permalinkParts;
+	let prefixElement, postNameElement, suffixElement;
+	if ( isEditable ) {
+		prefixElement = prefix && (
+			<span className="edit-post-post-link__link-prefix">{ prefix }</span>
+		);
+		postNameElement = postName && (
+			<span className="edit-post-post-link__link-post-name">{ postName }</span>
+		);
+		suffixElement = suffix && (
+			<span className="edit-post-post-link__link-suffix">{ suffix }</span>
+		);
+	}
+
+	const singularLabel = get( postType, [ 'labels', 'singular_name' ] );
+	return (
+		<PanelBody
+			title={
+				// translators: %s: post type singular name label e.g: Post, Page etc...
+				sprintf( __( '%s Link' ), singularLabel )
```

Comment:
I don't think we're allowed to do this in terms of i18n. right @ocean90 @swissspidy ?

Reply:
Correct, I'd use the generic `Permalink` as the title.